### PR TITLE
Remove TemplatingHelper

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Once this PR is merged the following files must be removed: 
- `app/bundles/CoreBundle/Templating/TemplateNameParser.php`
- `app/bundles/CoreBundle/Templating/TemplateReference.php`
- `templating.name_parser` from config

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Use themes in preference center, page and check the theme loads as expected

### CODE REVIEW

File: plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php [hunk 1]
```diff
@@ -283,17 +283,10 @@ function ($a, $b) {
 
     private function checkForMjmlTemplate($template)
```
Comment (mabumusa1) [inline plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php:287]: @escopecz this function is used in this [Controller](https://github.com/mautic/mautic/blob/5.x/plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php#L90) according to the code this function should not be used.

File: app/bundles/CoreBundle/Helper/ThemeHelper.php [hunk 2]
```diff
@@ -465,33 +453,36 @@ public function zip($themeName)
...
 
-            // Theme name is stored in the controller parameter
-            $template->set('controller', $theme['key']);
+            $fallbackTemplate = '@themes/'.$theme['key'].'/'.$templatePath;
 
-            if ($templating->getLoader()->exists($template)) {
-                return;
+            if ($this->twig->getLoader()->exists($template)) {
+                return $fallbackTemplate;
             }
```
Comment (mabumusa1) [inline app/bundles/CoreBundle/Helper/ThemeHelper.php:485]: @escopecz here is an extreme case, what should be rendered if the theme does not exist, the default theme does not exist or there is no fallback? I do not think we have built-in views for that, or do we? 

This is why PHPStan is read

Comment (escopecz) [reply app/bundles/CoreBundle/Helper/ThemeHelper.php:485]: We can fallback to the original behavior:
```php
public function getLogicalName()
    {
        return sprintf('%s:%s:%s.%s.%s', $this->parameters['bundle'], $this->parameters['controller'], $this->parameters['name'], $this->parameters['format'], $this->parameters['engine']);
    }
```
Or throw an exception?

File: plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php (hunk 1 above)
Comment (escopecz) [reply plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php:287]: It should be marked as deprecated then?

File: app/bundles/CoreBundle/Helper/ThemeHelper.php (hunk 2 above)
Comment (mabumusa1) [reply app/bundles/CoreBundle/Helper/ThemeHelper.php:485]: I will throw an exception, the original behavior will result in a broken result

File: plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php (hunk 1 above)
Comment (mabumusa1) [reply plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php:287]: I just marked it

File: app/bundles/CoreBundle/Helper/ThemeHelper.php [hunk 3]
```diff
@@ -463,37 +451,41 @@ public function zip($themeName)
...
         }
+
+        throw new BadConfigurationException(sprintf('Could not find theme %s nor a fail back theme to replace it', $requestedThemeName));
```
Comment (escopecz) [inline app/bundles/CoreBundle/Helper/ThemeHelper.php]:
```suggestion
        throw new BadConfigurationException(sprintf('Could not find theme %s nor a fall back theme to replace it', $requestedThemeName));
```

**Review by escopecz — CHANGES_REQUESTED** [review]
If I create a new email, select a theme and open the builder then I get 500 error in the ajax call with
```
Unable to find template \u0022GrapesJsBuilder\/Builder\/template.html.twig\u0022 (looked into: \/var\/www\/html\/vendor\/knplabs\/knp-menu\/src\/Knp\/Menu\/Resources\/views, \/var\/www\/html\/vendor\/symfony\/twig-bridge\/Resources\/views\/Form).
```

**Comment by escopecz** [comment]
@mabumusa1 seems that https://github.com/mautic/mautic/blob/5.x/plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php#L135 is missing `@`

**Comment by mabumusa1** [comment]
> @mabumusa1 seems that https://github.com/mautic/mautic/blob/5.x/plugins/GrapesJsBuilderBundle/Controller/GrapesJsController.php#L135 is missing `@`

Fixed in the last commit

**Review by escopecz — APPROVED** [review]
Thanks! I see no more issues in the code nor by manual testing 👍

**Pull request merged**